packages/react/src/componentsV2/addressSection/index.tsx

```diff
@@ -0,0 +1,179 @@
+    city?: string;
+    streetNumber?: string;
+    street?: string;
+  };
+  setFieldValue: (field: string, value: string | number) => void;
+  areaCodeValue: string;
+  countryCode?: string;
+};
+
+const AddressSection = ({
+  values,
+  setFieldValue,
+  areaCodeValue,
+  countryCode,
+}: AddressSectionProps) => {
+  const { addressList, findAddress, retrieveAddress, addressDetails } =
+    useAddressProvider();
+
+  const {
+    isOpen = true,
+    openMenu,
+    getMenuProps,
+    getToggleButtonProps,
+    getItemProps,
+    closeMenu,
+  } = useSelect({ items: addressList ?? [] });
+
+  const onChangePostcode = async (areaCode: string) => {
+    if (countryCode != null) {
+      await findAddress(areaCode, countryCode);
+      openMenu();
+    }
+  };
+
+  const handleRetrieveAddress = async (addressId: string) => {
+    if (countryCode != null) {
+      await retrieveAddress(addressId, countryCode);
+    }
+  };
+
+  useEffect(() => {
+    if (addressDetails != null) {
+      setFieldValue('deliveryAddress', addressDetails.addressLineOne ?? '');
+      setFieldValue('city', addressDetails.city ?? '');
+      setFieldValue('streetNumber', addressDetails.streetNumber ?? '');
+      setFieldValue('street', addressDetails.street ?? '');
+    }
+  }, [addressDetails, setFieldValue]);
+
+  return (
+    <StyledWrapper>
+      <div className="relative col-span-3 mb-2">
+        <StyledAreaCode>
+          <StyledInput>
+            <Input
+              fieldId="areaCode"
+              fieldName="areaCode"
+              setFieldValue={setFieldValue}
+              required
+              placeholder="Postal/Zip code"
+              label="Postal/Zip code"
+              onFocus={() =>
+                addressList && addressList.length > 0 && openMenu()
+              }
+            />
+          </StyledInput>
+
+          <StyledButton>
+            <Button
+              {...getToggleButtonProps()}
+              onClick={() => onChangePostcode(areaCodeValue)}
+              type="button"
+            >
+              <SearchIcon />
+            </Button>
+          </StyledButton>
+        </StyledAreaCode>
+        <StyledSelectDropdownContainer {...getMenuProps()}>
+          {isOpen && (
+            <StyledSelectDropdown>
+              {addressList?.map((item, index) => {
+                const label = item.labels.join(', ');
+                return (
+                  <StyledSelectItem
+                    {...getItemProps({ item, index })}
+                    key={`${item.id}`}
+                    {...getItemProps({
+                      item,
+                      index,
+                    })}
+                    onClick={() => {
+                      handleRetrieveAddress(item.id);
+                      closeMenu();
+                    }}
+                  >
+                    <div>
+                      {label.length < 45
+                        ? label
+                        : `${label.substring(0, 45)}...`}
```

Comment:
We have a service for this in dash, stringshorten or something. Maybe you can also make this a service in React button